src/Workspaces/Core/Portable/Shared/Utilities/SignatureComparer.cs

```diff
@@ -26,7 +26,7 @@ private SignatureComparer(SymbolEquivalenceComparer symbolEquivalenceComparer)
         public bool HaveSameSignature(ISymbol symbol1, ISymbol symbol2, bool caseSensitive)
         {
             // NOTE - we're deliberately using reference equality here for speed.
-            if (symbol1 == symbol2)
+            if ((object)symbol1 == symbol2)
```

Comment:
>(object) [](start = 16, length = 8)

The added cast is unnecessary, please revert.

Reply:
> The added cast is unnecessary, please revert.

FWIW, i disagree here.  The main problem is that it's very easy for someone to *accidentally* use reference-equality in C# with symbols.  I've seen this bite people many times in the compiler codebase.  This can def happen when the code started with `Symbol` and then moved to `ISymbol`.  All of a sudden, an call to an overloaded `==` becomes a ref-equals check without people realizing it.  This can lead to subtle bugs that go unnoticed.

Similarly, if just == is used, then a conversion from ISymbol to Symbol equivalently changes meaning.  

The main issue here is that it's not normally clear from the source what the actual *intent* was when using `==`.  Yes, the code and compiler are doing precisely what was written, but what was written may not have been what was *intended*.  To that end, it's often good to have the code be more self-documenting in terms of intent.  When you have something like `(object)x == y` or `ReferenceEquals(x, y)` then those forms make it very clear what the intent was, and both help avoid subtle bugs from creeping in with not-uncommon code changes.  

While the cahnge to use `(object)x` has no actual change to semantics, it very much helps with the code documentation and maintenance side of things, and i think that's a valuable thing for us to consider, esp. since we've had such bugs crop up due to this sort of thing in the past.